### PR TITLE
feature(hardware-controller): add the basic framework for error handling

### PR DESCRIPTION
**Pull request by ryanthecoder**
[RET-1244](https://opentrons.atlassian.net/browse/RET-1244)
[RET-1245](https://opentrons.atlassian.net/browse/RET-1245)
[RET-1246](https://opentrons.atlassian.net/browse/RET-1246)
[RET-1247](https://opentrons.atlassian.net/browse/RET-1247)

Add a message index to all of the can messages and generate unique Index for each packet
create an error message definition

# Overview

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: hardware/opentrons_hardware/drivers/can_bus/can_messenger.py [hunk 1]
```diff
@@ -54,11 +58,17 @@ def __init__(self, driver: AbstractCanDriver) -> None:
     async def send(self, node_id: NodeId, message: MessageDefinition) -> None:
         """Send a message."""
         # TODO (amit, 2021-11-05): Use function code when it is better defined.
```
Comment (Laura-Danielle) [inline hardware/opentrons_hardware/drivers/can_bus/can_messenger.py]: We can remove this TODO now.

File: hardware/opentrons_hardware/firmware_bindings/messages/message_definitions.py [hunk 2]
```diff
@@ -1,21 +1,70 @@
...
 
 from typing_extensions import Literal
 
 from ..constants import MessageId
 from . import payloads
+from .. import utils
+
+
+class SingletonMessageIndexGenerator(object):
+    """Singlton class that generates uinque index values."""
```
Comment (fsinapi) [inline hardware/opentrons_hardware/firmware_bindings/messages/message_definitions.py]: typo "SInglton"

**Review by Laura-Danielle — APPROVED** [review]
looks good to me, just one comment about removing a stale TODO

**Review by fsinapi — APPROVED** [review]
This looks really good! Just found one little typo

**Pull request merged**